packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -106,125 +36,23 @@ function calculateContent(trace: Trace, span: Span, allSpans: Span[], resultValu
-          const allChildrenWithout = _.difference(allChildren, longerAsParentSpan);
-          tempSelf = earliestLongerAsParent.relativeStartTime - span.relativeStartTime;
-          for (let i = 0; i < allChildrenWithout.length; i++) {
-            tempSelf -= allChildrenWithout[i].duration;
-          }
-        } else {
-          const overlayOnly = _.difference(allOverlay, longerAsParentSpan);
-          const allChildrenWithout = _.difference(allChildren, longerAsParentSpan);
-          const earliestLongerAsParent = _.minBy(longerAsParentSpan, function calc(a) {
-            return a.relativeStartTime;
-          });
-
-          // overlay between longerAsParent and overlayOnly
-          const overlayWithout = [];
-          for (let i = 0; i < overlayOnly.length; i++) {
-            if (!earliestLongerAsParent.relativeStartTime <= overlayOnly[i].relativeStartTime) {
-              overlayWithout.push(overlayOnly[i]);
-            }
-          }
-          for (let i = 0; i < overlayWithout.length; i++) {
-            if (
-              overlayWithout[i].relativeStartTime + overlayWithout[i].duration >
-              earliestLongerAsParent.relativeStartTime
-            ) {
-              overlayWithout[i].duration -=
-                overlayWithout[i].relativeStartTime +
-                overlayWithout[i].duration -
-                earliestLongerAsParent.relativeStartTime;
-              if (overlayWithout[i].duration < 0) {
-                overlayWithout[i].duration = 0;
-              }
-            }
-          }
-
-          tempSelf = onlyOverlay(overlayWithout, allChildrenWithout, tempSelf, span);
-          const diff = span.relativeStartTime + span.duration - earliestLongerAsParent.relativeStartTime;
-          tempSelf = Math.max(0, tempSelf - diff);
-        }
-      } else if (longerAsParent) {
-        // span is longer as Parent
-        tempSelf = longerAsParentSpan[0].relativeStartTime - span.relativeStartTime;
-        for (let i = 0; i < allChildren.length; i++) {
-          if (allChildren[i].spanID !== longerAsParentSpan[0].spanID) {
-            tempSelf -= allChildren[i].duration;
-          }
-        }
-      } else {
-        // Overlay
-        tempSelf = onlyOverlay(allOverlay, allChildren, tempSelf, span);
-      }
-    }
-    // no children
+    const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
+    // Filter for CHILD_OF we don't want to calculate FOLLOWS_FROM
+    const children = allSpans.filter(
+      each =>
+        span.childSpanIds.includes(each.spanID) &&
+        each.references[0].spanID === span.spanID &&
+        each.references[0].refType === 'CHILD_OF'
+    );
+    children.forEach(child => {
+      spanRange.subtract(10 * child.startTime, 10 * (child.startTime + child.duration) - 1);
+    });
+    const selfTime = spanRange
+      .subranges()
+      .map(r => r.high - r.low)
+      .reduce((sum, v) => sum + v, 0);
```

Comment:
This simpler form might work now:
```suggestion
    const selfTime = spanRange.length / 10;
```

Reply:
in DRange , length = high - low +1;

if spanRange has more than 10 subranges then there will additional 10 micro if we do spanRange.length.
Then we get extra 1 micro for selfTime.

May be we should keep the code same or we can do like this

`
const selfTime = (spanRange.length - spanRange.subranges().length)/10;`

Reply:
> in DRange , length = high - low +1;

but that's exactly the formula that we want. If we start with a span [10, 20] (duration = 10), convert to drange as [100, 199], then length will be 199 - 100 + 1 = 100, 100/10 = 10.

Reply:
Yeah! got it